### DISCUSSION TITLE
UIButtonのアニメーションを消したい

### DISCUSSION THREAD

**Discussion by AtsushiHosaka** [lit-iphone-course/lit-iphone-course#100] [Question-original] [answered]
# 質問
カウントアプリのアレンジを作っています。
UIButtonはタップされた時にすこしテキストの色が薄くなるアニメーションがデフォルトでついていると思うのですが、これを消したいです。
![スクリーンショット 2023-01-31 12 15 53](https://user-images.githubusercontent.com/26042964/215654077-b000f41f-85be-4691-bf4c-d55c1d84028c.png)

# やったこと
setTitleでbuttonのテキストを変更しているのですが、その際にフォントが変更されてしまったため、storyboardからstyleをplain→defaultと変更しました。
plainの状態でもアニメーションはありましたが、defaultに変更した際によりアニメーションが強くなったように感じます。（このフォントのことに関してもやり方が合っているかよく分かっていません…）
![スクリーンショット 2023-01-31 12 17 22](https://user-images.githubusercontent.com/26042964/215654245-a03c6d62-01da-46e3-ace8-0e17ed326624.png)

# 試したこと
・.removeMotionEffect（.motionEffectsが[]だったので効果なし）
・styleをplainのまま、.titleLabel?.font = UIFont("")（うまくいかず）

# 現在のプロジェクト
https://github.com/AtsushiHosaka/la5_Count_arrange

**Comment by txxxxc** [comment, marked as answer]
>UIButtonはタップされた時にすこしテキストの色が薄くなるアニメーションがデフォルトでついていると思うのですが、これを消したいです。

TypeをCustomにして、StyleをDefaultにすると消せるっぽいので試してみてほしいですー！
![image](https://user-images.githubusercontent.com/22112440/215672779-369d1048-80c5-45c4-b680-55a887f57936.png)

**Comment by AtsushiHosaka** [comment]
うまく行きました！ありがとうございます！
